elephant/current_source_density/KCSD.py

```diff
@@ -0,0 +1,1274 @@
+        **kwargs
+            configuration parameters, that may contain the following keys:
+            src_type : str
+                basis function type ('gauss', 'step', 'gauss_lim')
+                Defaults to 'gauss'
+            sigma : float
+                space conductance of the medium
+                Defaults to 1.
+            n_src_init : int
+                requested number of sources
+                Defaults to 300
+            R_init : float
+                demanded thickness of the basis element
+                Defaults to 0.23
+            h : float
+                thickness of analyzed cylindrical slice
+                Defaults to 1.
+            xmin, xmax : floats
+                boundaries for CSD estimation space
+                Defaults to min(ele_pos(x)), and max(ele_pos(x))
+            ext_x : float
+                length of space extension: x_min-ext_x ... x_max+ext_x
+                Defaults to 0.
+            gdx : float
+                space increments in the estimation space
+                Defaults to 0.01(xmax-xmin)
+            lambd : float
+                regularization parameter for ridge regression
+                Defaults to 0.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        LinAlgException
+            If the matrix is not numerically invertible.
+        KeyError
+            Basis function (src_type) not implemented. See basis_functions.py for available
+        """
+        super(KCSD1D, self).__init__(ele_pos, pots, **kwargs)
+        return
+
+    def estimate_at(self):
+        """Defines locations where the estimation is wanted
+        Defines:         
+        self.n_estm = self.estm_x.size
+        self.ngx = self.estm_x.shape
+        self.estm_x : Locations at which CSD is requested.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        nx = (self.xmax - self.xmin)/self.gdx
+        self.estm_x = np.mgrid[self.xmin:self.xmax:np.complex(0,nx)]
+        self.n_estm = self.estm_x.size
+        self.ngx = self.estm_x.shape[0]
+        return
+
+    def place_basis(self):
+        """Places basis sources of the defined type.
+        Checks if a given source_type is defined, if so then defines it
+        self.basis, This function gives locations of the basis sources, 
+        Defines
+        source_type : basis_fuctions.basis_1D.keys()
+        self.R based on R_init
+        self.dist_max as maximum distance between electrode and basis
+        self.nsx = self.src_x.shape
+        self.src_x : Locations at which basis sources are placed.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+        """
+        # check If valid basis source type passed:
+        source_type = self.src_type
+        try:
+            self.basis = basis.basis_1D[source_type]
+        except:
```

Comment:
Can you be more specific with the exception error, so what type so expect to catch? To catch everything is not a "good" style.